pkg/speccheck/builtin/process_service_instance.go

```diff
@@ -42,3 +67,74 @@ func (serviceInstance) ApplySpec(ctx *speccheck.Context, spec, actual *unstructu
 
 	return &instanceSpec, nil
 }
+
+// works around the fact that service catalog does not know when to send an update request if the
+// parameters section would change as the result of changing a secret referenced in parametersFrom. To
+// do this we record the contents of all referenced secrets in an annotation, compare that annotations value
+// each time a service instance is processed, and force UpdateRequests to a higher value when those secrets
+// change to trigger service catalog to send an update request.
+//
+// The annotation is only ever added to the spec object not the actual object. The modified spec is treated as
+// if the user manually set the annotation.
+//
+// If the a user actually did set the annotation manually there are two cases.
+// 1. if there are no parametersFrom referenced secrets we leave the spec untouched
+// 2. if there are referenced secrets we proceed to check and generate a new value for the annotation
+//
+// This can be disabled by adding the annotation with the value 'disabled'
+func (s serviceInstance) setSecretParametersChecksumAnnotation(ctx *speccheck.Context, spec, actual *sc_v1b1.ServiceInstance) error {
```

Comment:
This logic is mostly left untouched, however, we now always set the annotation regardless of whether there are secret references or not. (need to update the comment above!)